src/uu/top/src/top.rs

```diff
@@ -0,0 +1,239 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::{thread::sleep, time::Duration};
+
+use clap::{arg, crate_version, ArgAction, ArgGroup, ArgMatches, Command};
+use picker::sysinfo;
+use prettytable::{format::consts::FORMAT_CLEAN, Row, Table};
+use sysinfo::{Pid, Users};
+use uucore::{
+    error::{UResult, USimpleError},
+    format_usage, help_about, help_usage,
+};
+
+const ABOUT: &str = help_about!("top.md");
+const USAGE: &str = help_usage!("top.md");
+
+mod field;
+mod picker;
+
+#[allow(unused)]
+#[derive(Debug)]
+enum Filter {
+    Pid(Vec<u32>),
+    User(String),
+    EUser(String),
+}
+
+#[derive(Debug)]
+struct Settings {
+    // batch:bool
+    filter: Option<Filter>,
+    width: Option<usize>,
+}
+
+impl Settings {
+    fn new(matches: &ArgMatches) -> Self {
+        let filter = if let Some(pidlist) = matches.get_many::<u32>("pid") {
+            Some(Filter::Pid(pidlist.cloned().collect::<Vec<_>>()))
+        } else if let Some(user) = matches.get_one::<String>("filter-any-user") {
+            Some(Filter::User(user.clone()))
+        } else {
+            matches
+                .get_one::<String>("filter-only-euser")
+                .map(|euser| Filter::EUser(euser.clone()))
+        };
```

Comment:
It looks a bit odd to use two different ways to accomplish the same. Maybe it would make sense to add another `if let`.
```rust
        } else if let Some(user) = matches.get_one::<String>("filter-any-user") {
            Some(Filter::User(user.clone()))
        } else if let Some(euser) = matches.get_one::<String>("filter-only-euser") {
            Some(Filter::EUser(euser.clone()))
        } else {
            None
        };
```
Another idea is to get rid of all the `if`s and use `map` and `or_else`.
```rust
        let filter = matches
            .get_many::<u32>("pid")
            .map(|pidlist| Filter::Pid(pidlist.cloned().collect()))
            .or_else(|| {
                matches
                    .get_one::<String>("filter-any-user")
                    .map(|user| Filter::User(user.clone()))
            })
            .or_else(|| {
                matches
                    .get_one::<String>("filter-only-euser")
                    .map(|euser| Filter::EUser(euser.clone()))
            });
```